libcontainer/nsenter/nsexec.c

```diff
@@ -560,6 +613,11 @@ void nsexec(void)
 	if (ensure_cloned_binary() < 0)
 		bail("could not ensure we are a cloned binary");
 
+	/* Get the log pipe */
+	logfd = logpipe();
+
+	write_log(DEBUG, "%s started", __FUNCTION__);
```

Comment:
I reckon we should prefix all log messages from the C code with `__FUNCTION__:__LINE__` -- so this should be done in `write_log` as well.

Reply:
It would also be really nice to redefine `bail` to use `write_log` and then `exit`.

Reply:
@cyphar We tried to do that but when we moved `__FUNCTION__:__LINE__` into the `write_log` function we started always getting the same line number (the one in the write_log function) which is pretty useless. Are you sure you still want `bail` to use `write_log`?

Reply:
If you make `write_log` a macro which wraps an underlying function then it'll work properly -- since `__FUNCTION__:__LINE__` will always be the same if you put it inside an actual function, but putting it in a macro means it'll expand in place of the invocation.

Reply:
Done